In the fiat package recipe patch out "-nowarn nointerfaces" when version="um"

Currently the [Spack package recipe for `fiat`](api-v2/spack_repo/access/nri/packages/fiat/package.py) applies 
```
    patch("intel_warnings_v110.patch", when="@0:1.1.0")
    patch("intel_warnings_v120.patch", when="@1.2.0:")
```
With Spack version 1.1, neither patch is applied in the case of `when="@um"`, so that 
```
spack install fiat@um target=x86_64_v4 %intel@2021.10.0
```
results in
```
==> No binary for fiat-um-mdk4gttjgvoi42o3uxsctq67rltcjcyr found: installing from source
==> Installing fiat-um-mdk4gttjgvoi42o3uxsctq67rltcjcyr [17/17]
==> No patches needed for fiat
==> fiat: Executing phase: 'cmake'
==> Error: ProcessError: Command exited with status 1:
    '/g/data/tm70/pcl851/spack/1.1/release/linux-x86_64/cmake-3.31.11-tvrh2bn62qu4sj76grodz4es673bpnm4/bin/cmake' '-G' 'Unix Makefiles' '-DCMAKE_INSTALL_PREFIX:STRING=/g/data/tm70/pcl851/spack/1.1/release/linux-x86_64_v4/fiat-um-mdk4gttjgvoi42o3uxsctq67rltcjcyr' '-DCMAKE_INSTALL_RPATH_USE_LINK_PATH:BOOL=ON' '-DCMAKE_INSTALL_RPATH:STRING=/g/data/tm70/pcl851/spack/1.1/release/linux-x86_64_v4/fiat-um-mdk4gttjgvoi42o3uxsctq67rltcjcyr/lib;/g/data/tm70/pcl851/spack/1.1/release/linux-x86_64_v4/fiat-um-mdk4gttjgvoi42o3uxsctq67rltcjcyr/lib64' '-DCMAKE_PREFIX_PATH:STRING=/g/data/tm70/pcl851/spack/1.1/release/linux-x86_64/cmake-3.31.11-tvrh2bn62qu4sj76grodz4es673bpnm4;/g/data/tm70/pcl851/spack/1.1/release/linux-x86_64/compiler-wrapper-1.0-spro2dgnjjawulhw7fxxrqskm73cf4dd;/g/data/tm70/pcl851/spack/1.1/release/linux-x86_64/ecbuild-3.12.0-ad5brxdhickn5u5eaofzxd5m572phpb7;/g/data/tm70/pcl851/spack/1.1/release/linux-x86_64/fckit-0.14.1-ejqlvi22qv3swpvwnmygeontwl7evg2e;/g/data/tm70/pcl851/spack/1.1/release/linux-x86_64/gmake-4.4.1-d5nd3lks3aornpidrxz2hbteka3uuk3r;/g/data/tm70/pcl851/spack/1.1/release/linux-x86_64/eckit-1.32.3-upx5crwud5k7k34pcgzfskywythkexgy;/g/data/tm70/pcl851/spack/1.1/release/linux-x86_64/ncurses-6.5-20250705-eg6hnkr5hpgc4a7t2ydhckiydh4zj6s3;/apps/intel-ct/wrapper;/apps/python3/3.11.7;/apps/openmpi/4.1.7' '-DCMAKE_BUILD_TYPE:STRING=RelWithDebInfo' '-DCMAKE_VERBOSE_MAKEFILE:BOOL=ON' '-DCMAKE_INTERPROCEDURAL_OPTIMIZATION:BOOL=OFF' '-DCMAKE_POLICY_DEFAULT_CMP0090:STRING=NEW' '-DCMAKE_FIND_USE_PACKAGE_REGISTRY:BOOL=OFF' '-DCMAKE_EXPORT_COMPILE_COMMANDS:BOOL=ON' '-DENABLE_OMP:BOOL=ON' '-DENABLE_MPI:BOOL=ON' '-DENABLE_FCKIT:BOOL=ON' '-DCMAKE_C_COMPILER:STRING=/apps/openmpi/4.1.7/bin/mpicc' '-DCMAKE_CXX_COMPILER:STRING=/apps/openmpi/4.1.7/bin/mpic++' '-DCMAKE_Fortran_COMPILER:STRING=/apps/openmpi/4.1.7/bin/mpif90' '/scratch/tm70/pcl851/tmp/spack-stage/spack-stage-fiat-um-mdk4gttjgvoi42o3uxsctq67rltcjcyr/spack-src'

1 error found in build log:
     73    -- Added C flag [-diag-disable=279]
     74    -- Performing Test fiat_C_FLAG_TEST_8
     75    -- Performing Test fiat_C_FLAG_TEST_8 - Success
     76    -- Added C flag [-diag-disable=11076]
     77    -- Performing Test fiat_Fortran_FLAG_TEST_1
     78    -- Performing Test fiat_Fortran_FLAG_TEST_1 - Failed
  >> 79    CMake Error at /g/data/tm70/pcl851/spack/1.1/release/linux-x86_64/ecbuild-3.12.0-ad5brxdhickn5u5eaofzxd5m572phpb7/share/ecbuild/cmake/ecbuild_log.cmake:190 (message):
     80      CRITICAL - Fortran compiler /apps/openmpi/4.1.7/bin/mpif90 does not
     81      recognise Fortran flag '-nowarn nointerfaces'
     82    Call Stack (most recent call first):
     83      /g/data/tm70/pcl851/spack/1.1/release/linux-x86_64/ecbuild-3.12.0-ad5brxdhickn5u5eaofzxd5m572phpb7/share/ecbuild/cmake/ecbuild_add_lang_flags.cmake:128 (ecbuild_critical)
     84      /g/data/tm70/pcl851/spack/1.1/release/linux-x86_64/ecbuild-3.12.0-ad5brxdhickn5u5eaofzxd5m572phpb7/share/ecbuild/cmake/ecbuild_add_fortran_flags.cmake:44 (ecbuild_add_lang_fla
           gs)
     85      src/fiat/CMakeLists.txt:31 (ecbuild_add_fortran_flags)

See build log for details:
  /scratch/tm70/pcl851/tmp/spack-stage/spack-stage-fiat-um-mdk4gttjgvoi42o3uxsctq67rltcjcyr/spack-build-out.txt
```

In contrast, `spack install fiat@1.1.0 target=x86_64_v4 %intel@2021.10.0` results in
```
==> Installing fiat-1.1.0-cydipxaobclxi4gfjdbu7c3xqslso2cg [21/21]
==> Fetching https://mirror.spack.io/_source-cache/archive/58/58354e60d29a1b710bfcea9b87a72c0d89c39182cb2c9523ead76a142c695f82.tar.gz
    [100%]  242.20 KB @  576.8 KB/s
==> Applied patch /g/data/tm70/pcl851/spack/1.1/access-spack-packages/spack_repo/access/nri/packages/fiat/intel_warnings_v110.patch
==> fiat: Executing phase: 'cmake'
==> fiat: Executing phase: 'build'
==> fiat: Executing phase: 'install'
==> fiat: Successfully installed fiat-1.1.0-cydipxaobclxi4gfjdbu7c3xqslso2cg
```

A simple fix is to add
```
    patch("intel_warnings_v110.patch", when="@um")
```
to the `fiat` Spack package recipe.

Also,
```
    version("um", branch="um", no_cache=True)
```
should include a Git reference (a tag or a commit hash).

--- a/spack_repo/access/nri/packages/fiat/package.py
+++ b/spack_repo/access/nri/packages/fiat/package.py
@@ -22,7 +22,7 @@ class Fiat(CMakePackage):
     license("Apache-2.0")
 
     version("main", branch="main", no_cache=True)
-    version("um", branch="um", no_cache=True)
+    version("um", branch="um", tag="1.4.1-um", no_cache=True)
     version("1.2.0", sha256="758147410a4a3c493290b87443b4091660b915fcf29f7c4d565c5168ac67745f")
     version("1.1.0", sha256="58354e60d29a1b710bfcea9b87a72c0d89c39182cb2c9523ead76a142c695f82")
     version("1.0.0", sha256="45afe86117142831fdd61771cf59f31131f2b97f52a2bd04ac5eae9b2ab746b8")
@@ -49,6 +49,7 @@ class Fiat(CMakePackage):
 
     patch("intel_warnings_v110.patch", when="@0:1.1.0")
     patch("intel_warnings_v120.patch", when="@1.2.0:")
+    patch("intel_warnings_v120.patch", when="@um")
 
     def cmake_args(self):
         args = [

